devlab/tests/test_verify_dst_functionality.py

```diff
@@ -0,0 +1,318 @@
+        if self.dst_cloud.novaclient.keypairs.list():
+            for key_p in self.dst_cloud.novaclient.keypairs.list():
+                key_p.delete()
+
+        self.release_fips_tenant()
+        self.update_floatip_neutron_quota(
+            self.fip_quota_neutron, TEST_TENANT_NAME)
+
+    def check_vm_ssh_access(self, vm, ip_addr, username, cmd):
+        with settings(host_string=ip_addr, user=username,
+                      key=config.private_key['id_rsa'],
+                      abort_on_prompts=True, connection_attempts=3,
+                      disable_known_hosts=True):
+            try:
+                run(cmd, shell=False)
+                msg = 'VM {name} with ip {addr} is accessible via ssh'
+                return msg.format(name=vm.name, addr=ip_addr), True
+            except SystemExit:
+                msg = 'VM {name} with ip {addr} is not accessible via ssh'
+                return msg.format(name=vm.name, addr=ip_addr), False
+
+    def check_cinder_volume_on_vm(self, vm, ip_addr, username, cmd):
+        with settings(host_string=ip_addr,
+                      user=username,
+                      key=config.private_key['id_rsa'],
+                      shell=config.ssh_vm_shell,
+                      abort_on_prompts=True,
+                      connection_attempts=3,
+                      disable_known_hosts=True):
+            try:
+                for command in cmd:
+                    sudo(command, shell=True)
+                msg = 'Cinder volume is OK and accessible on' \
+                      'VM with name {name} and ip: {addr}'
+                return msg.format(name=vm.name, addr=ip_addr), True
+            except SystemExit:
+                msg = 'Cinder volume access FAIL on' \
+                      'VM with name {name} and ip: {addr}'
+                return msg.format(name=vm.name, addr=ip_addr), False
+
+    def release_fips_tenant(self):
+        fip_tenant_ids = [f['id'] for f in
+                          self.dst_cloud.neutronclient.list_floatingips(
+        )['floatingips']
+            if f['tenant_id'] == self.dst_tenant_id]
+
+        for f_id in fip_tenant_ids:
+            self.dst_cloud.neutronclient.delete_floatingip(f_id)
+        time.sleep(3)
+
+    def update_floatip_neutron_quota(self, quota_value, tenant_name):
+        try:
+            # update tenant quota block
+            self.dst_cloud.switch_user(user=self.dst_cloud.username,
+                                       password=self.dst_cloud.password,
+                                       tenant='admin')
+            self.dst_cloud.neutronclient.update_quota(
+                self.dst_tenant_id, {'quota': {'floatingip': quota_value}})
+            self.dst_cloud.switch_user(user=self.dst_cloud.username,
+                                       password=self.dst_cloud.password,
+                                       tenant=tenant_name)
+        except (KeystoneClientBaseException, NovaClientBaseException,
```

Comment:
Why here is excepted CinderClientBaseException, GlanceClientBaseException and GlanceClientBaseException?
As i see here can be raised only KeystoneClientBaseException and  NeutronClientException.